### ISSUE TITLE
Logs Volume Coloring not happening only for Verbose and Information

### ISSUE THREAD

**Issue by spanhotra** [grafana/grafana#61588]
**What happened**:

In the Grafana Loki viewer, only the log level Information and Verbose and not being colored in the "Logs volume" section.
But in the details, it is getting colored appropriately. 
![image](https://user-images.githubusercontent.com/79205065/212762584-91507ebb-6451-4bde-ba29-b0e23bfdf6d8.png)

**What you expected to happen**:

I expected the same coloring scheme to be applied in the logs volume section

**How to reproduce it (as minimally and precisely as possible)**:

1. Generate logs using serilog
2. Ingest using OpenTelemetry 
3. Export to Loki endpoint 

**Anything else we need to know?**:

I am trying to ingest logs using Open Telemetry exported for Loki, and logs are being generated using Serilog

**Environment**:
- Grafana version: v9.3.2
- Data source type & version: 
- OS Grafana is installed on: Podman Container
- User OS & Browser: Windows 10, Chrome
- Grafana plugins: NA 
- Others: NA

**Comment by gabor** [comment]
hi @spanhotra , could you tell us which log-level-names have different color in logs-volume compared to logs-details? thanks!

**Comment by spanhotra** [comment]
Hi @gabor 
Information and Trace levels have gray color, rest all are coming fine.

**Comment by gabor** [comment]
hi @spanhotra thanks for the info. i tried it in grafana9.3.2, and it seems to work for me, see screenshot:
<img width="290" alt="level" src="https://user-images.githubusercontent.com/51989/214067625-d948d993-d4ab-448e-92b4-cc37ca6daced.png">

could you provide the LogQL query please (please remove anything secret/private from it).

also, could you check this:
- find a log-line that has the right color in the logs-details, but it's gray in the logs-volume
- open the log-line, look at the details. is there a label named `level` with the correct value?
- if there is no such label, could you show us the log-line? (please remove anything secret/private from it)

the way the logs-lines figure out their "color" is slightly different from logs-volume: if there is no "level" label, the logs-volume becomes gray. but, logs-lines try to somehow parse the log-line and extract the log-level from it somehow.
your best choice is to make sure the loki-label "level" gets filled out with the log-level, then both system show the same colors.

**Comment by spanhotra** [comment]
Hi @gabor 

Below are the details:

Fatal Log line: [showing correct colors in both the sections]

![image](https://user-images.githubusercontent.com/79205065/214074922-6e8faeee-be8d-41be-ad5c-73175ca26019.png)

I can see the level log label as well:

![image](https://user-images.githubusercontent.com/79205065/214075099-e5679099-8fbd-494c-9892-cb41d1adea36.png)


For Information log line: [Gray color]

![image](https://user-images.githubusercontent.com/79205065/214075273-98df0e33-1437-4d8a-b096-bd4c34d4002e.png)

Although, I do not see the level log label:

![image](https://user-images.githubusercontent.com/79205065/214075479-c8d68b71-5dec-4a91-9fe2-28ef97caa459.png)

All the logs are getting process the exact same way, not sure the difference,

**Comment by gabor** [comment]
@spanhotra thanks for the info.
you see, the field "severity" in the log-line-text causes the line to get the correct color.

what i recommend, is to tell loki to:
1. parse the log-line's JSON data ( this will create a loki-label named "serverity")
2. rename the severity-label to level-label

this should ensure that you always have the loki-label "level" with a good value.
if your logql query looks like `{....}`, then it will become `{...} | json | label_format level=severity`

could you try that?

**Comment by spanhotra** [comment]
Hi @gabor 

`{...} | json | label_format level=severity` worked!!!

But renaming did not work, as it started to merge the provided values with the values Loki extracted. and filtering over it is not working. 

Is it related to Serilog logs being ingested with Otel Collector instead of Promtail?

**Comment by gabor** [comment]
> Hi @gabor
> 
> `{...} | json | label_format level=severity` worked!!!

i'm glad 👍 

> 
> But renaming did not work, as it started to merge the provided values with the values Loki extracted. and filtering over it is not working.

i'm sorry, could you explain this part in more detail?

> 
> Is it related to Serilog logs being ingested with Otel Collector instead of Promtail?

i'm not sure about what the best ingesting approach is, asking @grafana/loki-squad , could you take a look at this issue please?

**Comment by spanhotra** [comment]
When I renamed using Otel Processor,  it merged the values.
Eg. in my log file if the values of severity were "Error", "Information", "Fatal", "Trace", "Warning"

The final filter list that got constructed in the Garafana UI was "Error", "Information", "Fatal", "Trace", "Warning", "ERROR", "WARNING", "FATAL"

So, it is still extracting level somehow for severity >= 'Warning' automatically.

**Comment by spanhotra** [comment]
Hi @gabor 

I figured out the issue, you were right all along, I pre processed the logs to extract the severity and map it to the correct terminology and it worked.

Here is the change I did,

![image](https://user-images.githubusercontent.com/79205065/214287630-80111d50-483e-4afa-ad55-397b2984b4d0.png)

For Critical and Verbose, the values were not matching with fatal and trace, hence Grafana was not able to infer the correct mapping, I had to explicitly give the mappings in my Otel Collector config. and it worked.

Thanks for bearing with me,
I am closing the issue.
